internal/controller/streams.go

```diff
@@ -426,24 +432,57 @@ func (sm *streamManager) StreamConfigurationUpdates(ctx context.Context, logger
 }
 
 // startFlowCacheOutReader starts a goroutine that reads flows from the flow cache and sends them to the cloud secure.
-func (sm *streamManager) startFlowCacheOutReader(ctx context.Context, logger *zap.Logger) error {
+func (sm *streamManager) startFlowCacheOutReader(ctx context.Context, logger *zap.Logger, networkFlowStreamOpen chan struct{}) error {
 	flowCount := 0
-	ticker := time.NewTicker(60 * time.Second) // Create a ticker that triggers every 60 seconds
-	defer ticker.Stop()                        // Ensure the ticker is stopped when the function exits
+	allowedFlows := 200                           // Initialize allowed flows to some default value IE 200, This value is set by the server in the ServerWindow message
+	logger.Info("Starting flow cache out reader") // Ensure the ticker is stopped when the function exits
+	ticker := time.NewTicker(60 * time.Second)    // Create a ticker that triggers every 60 seconds
+	defer ticker.Stop()                           // Ensure the ticker is stopped when the function exits
+	// Goroutine to read ServerWindow messages from the network flow stream
+	go func() {
+		<-networkFlowStreamOpen // Wait for network flow stream to be open
+		for {
+			resp, err := sm.streamClient.networkFlowsStream.Recv()
+			if err == io.EOF {
+				logger.Info("Server has closed the network flows stream")
+				return
+			}
+			if err != nil {
+				logger.Error("Error receiving response from network flows stream", zap.Error(err))
+				return
+			}
+
+			if serverWindow, ok := resp.Response.(*pb.SendKubernetesNetworkFlowsResponse_ServerWindow); ok {
+				allowedFlows = int(serverWindow.ServerWindow.AllowedMessages)
+				logger.Debug("Received ServerWindow message", zap.Int("allowed_flows", allowedFlows))
+				flowCount = 0 // Reset flowCount when a new ServerWindow message is received
```

Comment:
Both `allowedFlows` and `flowCount` are read in another goroutine, so all reads and writes to both variables must be protected by a mutex.